textworld/utils.py

```diff
@@ -192,6 +193,23 @@ def uniquify(seq):
     return [x for x in seq if x not in seen and not seen.add(x)]
 
 
+def take(n: int, iterable: Iterable) -> Iterable:
+    """ Return first n items of the iterable as a list.
+
+    References:
+        https://docs.python.org/3/library/itertools.html#itertools-recipes
+    """
+    return list(itertools.islice(iterable, n))
+
+
+def chunk(iterable: Iterable, n: int) -> Iterable[Iterable]:
```

Comment:
You could avoid building the intermediate lists with something like:

```
it = iter(iterable)
while True:
    first = next(it)
    rest = itertools.islice(it, n - 1)
    yield itertools.chain((first,), rest)
    # Make sure rest is exhausted
    for _ in rest:
        pass
```

Reply:
Actually, from a user perspective, I expect chunk to return a list of elements (I'll change the return type to `Iterable[List]`. But I like your version, good thinking for the `for _ in rest: pass` is a nice failsafe. I think I'll with

```python
def chunk(iterable: Iterable, n: int, fct: Callable = lambda e: e) -> Iterable[Iterable]:
    it = iter(iterable)
    while True:
        try:
            first = next(it)
        except StopIteration:
            return

        rest = itertools.islice(it, n - 1)
        yield fct(itertools.chain((first,), rest))
        # Make sure rest is exhausted
        for _ in rest:
            pass
```
which doesn't raise a "DeprecationWarning: generator 'chunk' raised StopIteration" and allows me to do
`list(chunk("asdasjdkhakjfhaskjfhaksjfdsfkjsdf", 10, "".join))`

Reply:
I think `fct` doesn't add that much value, you could always just do `map("".join, chunk("...", 10))` if you need it